src/imu/mpu9255/mpu9255.cpp

```diff
@@ -0,0 +1,150 @@
+    gyroscope_ = std::make_unique<imu::Gyroscope>(i2c_handler_);
+  }
+  if (accelerometer_ == NULL) {
+    accelerometer_ = std::make_unique<imu::Accelerometer>(i2c_handler_);
+  }
+  if (magnetometer_ == NULL) {
+    magnetometer_ = std::make_unique<imu::Magnetometer>(i2c_handler_);
+  }
+  if (temperature_ == NULL) {
+    temperature_ = std::make_unique<imu::Temperature>(i2c_handler_);
+  }
+}
+
+auto Mpu9255::SetInitConfigMPU9255(void) noexcept -> void {
+  SetMPU9255Register(INT_PIN_CFG, 0x22);
+  SetMPU9255Register(INT_ENABLE, 0x01);
+}
+
+auto Mpu9255::SetMPU9255Register(const std::uint8_t register_, const std::uint8_t register_value) noexcept -> void {
+  i2c_handler_->Write(MPU9255_ADDRESS, {register_, register_value});
+}
+
+auto Mpu9255::InitAllSensors(void) noexcept -> bool {
+  if (gyroscope_->Init(MPU9255_ADDRESS) == types::DriverStatus::OK &&
+      accelerometer_->Init(MPU9255_ADDRESS) == types::DriverStatus::OK &&
+      magnetometer_->Init(AK8963_ADDRESS) == types::DriverStatus::OK &&
+      temperature_->Init(MPU9255_ADDRESS) == types::DriverStatus::OK) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+auto Mpu9255::Update(void) noexcept -> types::DriverStatus {
+  if (!IsInitialized())
+    return types::DriverStatus::HAL_ERROR;
+
+  if (UpdateAllSensors()) {
+    initialized_ = true;
+    return types::DriverStatus::OK;
+  }
+  return types::DriverStatus::HAL_ERROR;
+}
+
+auto Mpu9255::IsInitialized(void) noexcept -> bool {
+  return initialized_;
+}
+
+auto Mpu9255::UpdateAllSensors(void) noexcept -> bool {
+  if (gyroscope_->Update() == types::DriverStatus::OK &&
+      accelerometer_->Update() == types::DriverStatus::OK &&
+      magnetometer_->Update() == types::DriverStatus::OK &&
+      temperature_->Update() == types::DriverStatus::OK) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+auto Mpu9255::SetGyroscopeSensitivity(const types::ImuSensitivity gyroscope_sensitivity) noexcept -> types::DriverStatus {
+  if (!IsInitialized())
+    return types::DriverStatus::HAL_ERROR;
+
+  return gyroscope_->SetSensitivity(gyroscope_sensitivity);
+}
+
+auto Mpu9255::GetGyroscopeSensitivity(void) noexcept -> types::ImuSensitivity {
+  if (!IsInitialized())
+    return types::ImuSensitivity::FINEST;
+
+  return gyroscope_->GetSensitivity();
+}
+
+auto Mpu9255::SetAccelerometerSensitivity(const types::ImuSensitivity accelerometer_sensitivity) noexcept -> types::DriverStatus {
+  if (!IsInitialized())
+    return types::DriverStatus::HAL_ERROR;
+
+  return accelerometer_->SetSensitivity(accelerometer_sensitivity);
+}
+
+auto Mpu9255::GetAccelerometerSensitivity(void) noexcept -> types::ImuSensitivity {
+  if (!IsInitialized())
+    return types::ImuSensitivity::FINEST;
+
+  return accelerometer_->GetSensitivity();
+}
+
+auto Mpu9255::GetGyroscope(void) noexcept -> types::EuclideanVector<std::int16_t> {
+  if (!IsInitialized())
+    return types::EuclideanVector<std::int16_t>{-1, -1, -1};
```

Comment:
Code duplication, the object EuclideanVector<std::int16_t>{-1,-1,-1} could be factored out and used in all three methods

Reply:
Correct. Add of ReturnVectorDefault to remove duplicated code.